lib/datadog/tracing/configuration/settings.rb

```diff
@@ -90,13 +90,25 @@ def self.extended(base)
               option :propagation_style do |o|
                 o.type :array
                 o.env Configuration::Ext::Distributed::ENV_PROPAGATION_STYLE
+                o.backup_env Configuration::Ext::Distributed::ENV_OTEL_PROPAGATION_STYLE
                 o.default []
                 o.after_set do |styles|
                   next if styles.empty?
 
                   # Make values case-insensitive
                   styles.map!(&:downcase)
 
+                  if effective_env(:propagation_style) == Configuration::Ext::Distributed::ENV_OTEL_PROPAGATION_STYLE
+                    styles.collect! do |s|
+                      s = 'b3' if s == 'b3single'
+                      if Configuration::Ext::Distributed::PROPAGATION_STYLE_SUPPORTED.include?(s)
+                        s
+                      else
+                        Datadog.logger.warn("Unsupported propagation style #{s} in #{Configuration::Ext::Distributed::ENV_OTEL_PROPAGATION_STYLE}")
+                        nil
+                      end
+                    end.compact!
+                  end
```

Comment:
How about a simpler version:
```suggestion
                  styles.select! do |s|
                    if Configuration::Ext::Distributed::PROPAGATION_STYLE_SUPPORTED.include?(s)
                      true
                    else
                      Datadog.logger.warn("Unsupported propagation style #{s}")
                      false
                    end
                  end
```